docs/UML/UML.puml

```diff
@@ -0,0 +1,164 @@
+@startuml
```

Comment:
Use a better name for this file please.

Reply:
I prefer merging contents first in case of further unexpected conflicts